### PR TITLE
mqtt: only use http-proxy() if it is supported

### LINKED ISSUE #3765: syslog-ng git snapshot does not compile on openSUSE after PR  #3747
syslog-ng git snapshot does not compile on openSUSE 15.2 after https://github.com/syslog-ng/syslog-ng/pull/3747 was merged. You can see the error message and info about the used MQTT package below:

```
[   68s] libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./lib -I./modules -I./lib -I./modules -I./modules/mqtt/destination/ -I./modules/mqtt/destination/ -pthread -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_28 -DGLIB_VERSION_MAX_ALLOWED=0x023C00 -I./lib/eventlog/src -I./lib/eventlog/src -DOPENSSL_LOAD_CONF -D_BSD_SOURCE -D__BSD_SOURCE -D__FAVOR_BSD -DHAVE_NET_ETHERNET_H -D_DEFAULT_SOURCE -I/usr/include/dbi -I/usr/include/dbi/dbi -isystem ./lib/ivykis/src/include -isystem ./lib/ivykis/src/include -D_GNU_SOURCE -D_DEFAULT_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wshadow -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros -O2 -g -pthread -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -c modules/mqtt/destination/mqtt-worker.c  -fPIC -DPIC -o modules/mqtt/destination/.libs/libmqtt_destination_la-mqtt-worker.o
[   68s] modules/mqtt/destination/mqtt-worker.c: In function '_connect':
[   68s] modules/mqtt/destination/mqtt-worker.c:232:16: error: 'MQTTClient_connectOptions {aka struct <anonymous>}' has no member named 'httpProxy'
[   68s]        conn_opts.httpProxy = owner->http_proxy;
[   68s]                 ^
[   68s] modules/mqtt/destination/mqtt-worker.c:233:16: error: 'MQTTClient_connectOptions {aka struct <anonymous>}' has no member named 'httpsProxy'
[   68s]        conn_opts.httpsProxy = owner->http_proxy;
[   68s]                 ^
[   68s] /bin/sh ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I.  -I./lib -I./modules -I./lib -I./modules  -I./modules/mqtt/destination/ -I./modules/mqtt/destination/ -pthread -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_28 -DGLIB_VERSION_MAX_ALLOWED=0x023C00 -I./lib/eventlog/src -I./lib/eventlog/src  -DOPENSSL_LOAD_CONF -D_BSD_SOURCE -D__BSD_SOURCE -D__FAVOR_BSD -DHAVE_NET_ETHERNET_H -D_DEFAULT_SOURCE -I/usr/include/dbi -I/usr/include/dbi/dbi -isystem ./lib/ivykis/src/include -isystem ./lib/ivykis/src/include  -D_GNU_SOURCE -D_DEFAULT_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wshadow -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros   -O2 -g -pthread -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -c -o modules/mqtt/destination/libmqtt_destination_la-mqtt-destination.lo `test -f 'modules/mqtt/destination/mqtt-destination.c' || echo './'`modules/mqtt/destination/mqtt-destination.c
[   68s] make[2]: *** [Makefile:19579: modules/mqtt/destination/libmqtt_destination_la-mqtt-worker.lo] Error 1
[   68s] make[2]: *** Waiting for unfinished jobs....
[   68s] libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./lib -I./modules -I./lib -I./modules -I./modules/mqtt/destination/ -I./modules/mqtt/destination/ -pthread -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_28 -DGLIB_VERSION_MAX_ALLOWED=0x023C00 -I./lib/eventlog/src -I./lib/eventlog/src -DOPENSSL_LOAD_CONF -D_BSD_SOURCE -D__BSD_SOURCE -D__FAVOR_BSD -DHAVE_NET_ETHERNET_H -D_DEFAULT_SOURCE -I/usr/include/dbi -I/usr/include/dbi/dbi -isystem ./lib/ivykis/src/include -isystem ./lib/ivykis/src/include -D_GNU_SOURCE -D_DEFAULT_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wshadow -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros -O2 -g -pthread -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -c modules/mqtt/destination/mqtt-destination.c  -fPIC -DPIC -o modules/mqtt/destination/.libs/libmqtt_destination_la-mqtt-destination.o
[   68s] make[1]: *** [Makefile:24330: all-recursive] Error 1
[   68s] make: *** [Makefile:10511: all] Error 2
```

and the MQTT package:

```
abuild@leap152:~> rpm -qi libpaho-mqtt-devel-1.3.1-lp152.1.4.x86_64
Name        : libpaho-mqtt-devel
Version     : 1.3.1
Release     : lp152.1.4
Architecture: x86_64
Install Date: Fri Jul 30 07:30:51 2021
Group       : Unspecified
Size        : 270618
License     : EPL-1.0 AND BSD-3-Clause
Signature   : RSA/SHA256, Sat May 16 14:59:53 2020, Key ID b88b2fd43dbdc284
Source RPM  : paho-mqtt-c-1.3.1-lp152.1.4.src.rpm
Build Date  : Sat May 16 14:59:42 2020
Build Host  : lamb06
Relocations : (not relocatable)
Packager    : https://bugs.opensuse.org
Vendor      : openSUSE
URL         : https://eclipse.org/paho/clients/c/
Summary     : Development files for MQTT C Client library
Description :
Development files for the the Paho MQTT C Client.
Distribution: openSUSE Leap 15.2
```

### PR DESCRIPTION
**Pull request by alltilla**
Fixes #3765

Alternative fix possibilities, that I could not make work:
 * I checked, whether we can use leb version related preprocessor macros from `libpaho-mqtt-dev`, but I could not find any.
 * `MQTTClient_connectOptions` has a `struct_version` member and `MQTTClient_connectOptions_initializer`, which sets it from a preprocessor macro. It seems like it could help, but I could not get useful information from it in compile time.

No news entry is needed, because the `http-proxy()` feature was not yet released.

Signed-off-by: Attila Szakacs <attila.szakacs@oneidentity.com>

### CODE REVIEW

**Review by MrAnno — DISMISSED** [review]
Thank you very much for fixing this!

File: configure.ac [hunk 1]
```diff
@@ -1330,6 +1330,24 @@ if test "x$enable_mqtt" = "xyes" || test "x$enable_mqtt" = "xauto"; then
...
+
+		AC_CHECK_MEMBER([MQTTClient_connectOptions.httpProxy],
+				[have_paho_http_proxy=1],
+				[have_paho_http_proxy=0],
+				[[#include "MQTTClient.h"]])
+
+		CPPFLAGS="$CPPFLAGS_SAVE"
+		LDFLAGS="$LDFLAGS_SAVE"
+
+		AC_DEFINE(HAVE_PAHO_HTTP_PROXY, $have_paho_http_proxy, [libpaho-mqtt supports MQTTClient_connectOptions::httpProxy])
```
[the remark this answers is not in the document]
Comment (alltilla) [reply configure.ac]: Wow, I don't know how I missed that. Thanks!

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by MrAnno — APPROVED** [review]

**Comment by czanik** [comment]
I did a fresh compile on openSUSE / SLES using this patch, and syslog-ng compiled successfully both on SLES and openSUSE Leap releases.

**Pull request merged**